### DISCUSSION TITLE
Unable to set QueryString in Custom Plugin access phase

### DISCUSSION THREAD

**Discussion by sivasubramani300890** [Kong/kong#9976] [Help] [answered]
Hi Team,
            I am trying to get query string using "kong.request.get_raw_query()" and after changing to right values setting the query string using "kong.service.request.set_raw_query(queryString)".But it is not happening.
            
            Any suggestion please

**Comment by samugi** [comment, marked as answer]
Hello @sivasubramani300890 please have a look at [this issue](https://github.com/Kong/kong/issues/9492), the comments should be helpful to resolve your problem ([this one](https://github.com/Kong/kong/issues/9492#issuecomment-1268085707) in particular).

**Reply by sivasubramani300890** [reply]
Hi @samugi, 
Thanks for your reply. I have used **python** to develop the third-party plugin. [https://github.com/Kong/kong/issues/9492] this issue is resolved in Lua it seems 

Even though I tried below code in third party python plugin

queryString = ngx.req.get_uri_args() 
kong.response.exit(200,queryString)

or 

queryString = kong.request.get_raw_query()
ngx.req.set_uri_args("foo=bar")

But both didn't worked.
Can you please advise if anything I missed in the above code?
Do I need to update anything in the library, please?

Note : I am using python to develop third party plugin

**Reply by samugi** [reply]
Hi @sivasubramani300890,
can you please try fetching the parameters with [kong.request.get_query](https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.request/#kongrequestget_querymax_args) and setting them via [kong.service.request.set_query()](https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.service.request/#kongservicerequestset_queryargs) or [kong.service.request.set_raw_query()](https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.service.request/#kongservicerequestset_queryargs)? If you encounter any issues please do share some additional details of the problem and the kong version you are testing this on.

**Reply by sivasubramani300890** [reply]
Hi @samugi , Thanks for the help.it's working
